test/aspects/ec2-automated-shutdown/index.test.ts

```diff
@@ -0,0 +1,430 @@
+import { Aspects, Stack, Duration } from 'aws-cdk-lib';
+import { Template, Match } from 'aws-cdk-lib/assertions';
+import {
+    Instance,
+    InstanceType,
+    InstanceClass,
+    InstanceSize,
+    AmazonLinuxImage,
+    Vpc,
+    SubnetType,
+    SecurityGroup,
+    BlockDeviceVolume
+} from 'aws-cdk-lib/aws-ec2';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { NagSuppressions } from 'cdk-nag';
+import {
+    Ec2AutomatedShutdown,
+    Ec2MetricName
+} from '../../../src/aspects/ec2-automated-shutdown';
+import { describeCdkTest } from '../../../utilities/cdk-nag-jest';
+import { mockVpcId, mockCidrBlock } from '../../fixtures/network';
+import { ComparisonOperator } from 'aws-cdk-lib/aws-cloudwatch';
+
+describeCdkTest(Ec2AutomatedShutdown, (_, getStack, getTemplate) => {
+    let stack: Stack;
+    let template: Template;
+
+    beforeEach(() => {
+        stack = getStack();
+        stack.node.setContext(
+            '@aws-cdk/aws-cloudwatch-actions:changeLambdaPermissionLogicalIdForLambdaAction',
+            true
+        );
+
+        NagSuppressions.addStackSuppressions(stack, [
+            {
+                id: 'AwsSolutions-IAM4',
+                reason: 'Not testing IAM in this test scenario.'
+            },
+            {
+                id: 'NIST.800.53.R5-CloudWatchLogGroupEncrypted',
+                reason: 'Mock function for testing'
+            },
+            {
+                id: 'AwsSolutions-EC28',
+                reason: 'Mock instance for testing; no autoscaling'
+            },
+            {
+                id: 'AwsSolutions-EC29',
+                reason: 'Mock instance for testing; no autoscaling'
+            },
+            {
+                id: 'AwsSolutions-L1',
+                reason: 'Lambda function is configured to use latest runtime version'
```

Comment:
It is currently set to 20 (NODEJS_20_X) this should be set to NODEJS_22_X to align with the rest